internal/config/settings.go

```diff
@@ -298,6 +299,22 @@ func WriteSettings(filename string) error {
 			}
 		}
 
+		// check for changes and prevent unnecessary writes
+		if len(parsedSettings) == len(oldParsedSettings) {
+			equal := true
+			for k, _ := range parsedSettings {
+				vNew, okNew := parsedSettings[k]
+				vOld, okOld := oldParsedSettings[k]
+				if !okNew || !okOld || !reflect.DeepEqual(vNew, vOld) {
+					equal = false
+					break
+				}
+			}
+			if equal {
+				return nil
+			}
+		}
```

Comment:
Instead of all this we can just use a local boolean variable which we set to true if we modify or delete anything in `parsedSettings` in this function?

...But I actually have doubts whether it's a good idea to prevent unnecessary writes so generically. Think of such a scenario, for example:

1. The user starts micro -> `settings.json` is read, option `foo` is set to true.
2. The user edits `settings.json`, changing `foo` to false (but doesn't run `reload`).
3. The user runs `set foo true` -> micro doesn't update `settings.json` (so the permanent `foo` setting remains false, i.e. the `set` command doesn't behave as it should).

Since micro unfortunately encourages both ways of setting settings (manual editing of `settings.json` and the `set` command), we should aim to ensure that both work well, together? So we shouldn't assume that `settings.json` hasn't changed since the last time micro read it, in the general case? We can only assume that in the special `reload` case?